provider/resource_corp_cloudwaf_instance.go

```diff
@@ -0,0 +1,341 @@
+					}
+					if existsInString(val.(string), "1.0", "1.2") {
+						return nil, nil
+					}
+					return nil, []error{errors.New(`tlsMinVersion must be "1.0" or "1.2"`)}
+				},
+			},
+			"use_uploaded_certificates": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"workspace_configs": {
+				Type:        schema.TypeSet,
+				Description: "Workspace Configs",
+				Required:    true,
+				MaxItems:    5,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"site_name": {
+							Type:        schema.TypeString,
+							Description: "Site name.",
+							Required:    true,
+						},
+						"instance_location": {
+							Type:        schema.TypeString,
+							Description: `Set instance location to "direct" or "advanced".`,
+							Required:    true,
+						},
+						"client_ip_header": {
+							Type:        schema.TypeString,
+							Description: `Specify the request header containing the client IP address, available when InstanceLocation is set to "advanced". Default: "X-Forwarded-For".`,
+							Optional:    true,
+						},
+						"listener_protocols": {
+							Type:        schema.TypeSet,
+							Description: "type: group - Conditions that must be matched when evaluating the request (all, any)",
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"routes": {
+							Type:        schema.TypeSet,
+							Description: "Routes",
+							Required:    true,
+							MaxItems:    200,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:        schema.TypeString,
+										Description: "Route unique identifier.",
+										Computed:    true,
+									},
+									"certificate_ids": {
+										Type:        schema.TypeSet,
+										Description: "List of certificate IDs in string associated with request URI or domains. IDs will be available in certificate GET request.",
+										Optional:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+									"origin": {
+										Type:        schema.TypeString,
+										Description: "Origin server URI.",
+										Required:    true,
+									},
+									"pass_host_header": {
+										Type:        schema.TypeBool,
+										Description: "Pass the client supplied host header through to the upstream (including the upstream TLS handshake for use with SNI and certificate validation). If using Heroku or Server Name Indications (SNI), this must be disabled(default: false).",
+										Optional:    true,
+									},
+									"domains": {
+										Type:        schema.TypeSet,
+										Description: "List of domain or request URIs, up to 100 entries.",
+										Required:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+									"connection_pooling": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"tls_host_override": {
```

Comment:
Should not be an issue since the field is optional but this field was recently removed from the API and no longer used.

Reply:
Could we also add trust_proxy_header as well?

Reply:
Thanks for the comments.

* `tls_host_override` has been removed
* `trust_proxy_headers` has been added